### ISSUE TITLE
panic: runtime error: index out of range [1] with length 1

### ISSUE THREAD

**Issue by j-mnr** [nfx/go-htmltable#3]
# :wave: Hey this looks like an awesome tool!

I needed something like this a couple of weeks ago and would love to see this working!

Unfortunately the first URL I tried resulted in a panic.

[playground link](https://goplay.tools/snippet/SUV2GtuNJCq)

Not sure if it was because there are multiple tables on the page? Or maybe I misunderstood the API, let me know if there is a better way I should set this up to get it working.

## Stack trace

```
2022/10/15 18:34:16 found table columns=[Model Code name Released CPU support Fab(nm) FSB/HT(MHz) Southbridge Features / Notes] count=7
2022/10/15 18:34:16 found table columns=[Model AMD480Xchipset(originally CrossFire Xpress 1600) Model RD480 Model Oct 2006 Model Athlon 64,Sempron Model 110 Model 1000(HT 2.0) Model No
 Model x8 + x8 Model SB600,ULi-M1575 Model ? Codename Released CPU support Fab(nm) HT(MHz) IGP CrossFire Southbridge Features / Notes] count=26
2022/10/15 18:34:16 found table columns=[Model Codename Released CPU support Fab(nm) HT(MHz) AMD-V(Hardware Virtualization) IGP CrossFire SLI[5] TDP(W) Southbridge Features / Notes] co
unt=13
2022/10/15 18:34:16 found table columns=[Model AMD 640 chipset Model AMD-645 Model 1997 Model  Model  Model  Model 2 × ATA/33 Model  Codename Released Fab(nm) USB2.0 + 1.1 Audio Parall
el ATA1 Features / Notes] count=14
2022/10/15 18:34:16 found table columns=[Model Codename Released Fab(nm) SATA USB2.0 + 1.1 Parallel ATA1 RAID NIC Package TDP(W) Features / Notes] count=12
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
github.com/nfx/go-htmltable.(*Page).finishTable(0xc000207d18)
        GOPATH/pkg/mod/github.com/nfx/go-htmltable@v0.2.1/page.go:273 +0x867
github.com/nfx/go-htmltable.(*Page).parse(0xc000207d18, 0xc0004e80e0)
        GOPATH/pkg/mod/github.com/nfx/go-htmltable@v0.2.1/page.go:219 +0x427
github.com/nfx/go-htmltable.(*Page).parse(0xc000207d18, 0xc00026bce0)
        GOPATH/pkg/mod/github.com/nfx/go-htmltable@v0.2.1/page.go:222 +0x45a
github.com/nfx/go-htmltable.(*Page).parse(0xc000207d18, 0xc00026bc70)
        GOPATH/pkg/mod/github.com/nfx/go-htmltable@v0.2.1/page.go:222 +0x45a
github.com/nfx/go-htmltable.(*Page).parse(0xc000207d18, 0xc00026b500)
        GOPATH/pkg/mod/github.com/nfx/go-htmltable@v0.2.1/page.go:222 +0x45a
github.com/nfx/go-htmltable.(*Page).parse(0xc000207d18, 0xc00026aee0)
        GOPATH/pkg/mod/github.com/nfx/go-htmltable@v0.2.1/page.go:222 +0x45a
github.com/nfx/go-htmltable.(*Page).parse(0xc000207d18, 0xc00026acb0)
        GOPATH/pkg/mod/github.com/nfx/go-htmltable@v0.2.1/page.go:222 +0x45a
github.com/nfx/go-htmltable.(*Page).parse(0xc000207d18, 0xc000226a10)
        GOPATH/pkg/mod/github.com/nfx/go-htmltable@v0.2.1/page.go:222 +0x45a
github.com/nfx/go-htmltable.(*Page).parse(0xc000207d18, 0xc000226930)
        GOPATH/pkg/mod/github.com/nfx/go-htmltable@v0.2.1/page.go:222 +0x45a
github.com/nfx/go-htmltable.(*Page).init(0x68d300?, {0x7f3d447d5830?, 0xc00049c180?})
        GOPATH/pkg/mod/github.com/nfx/go-htmltable@v0.2.1/page.go:186 +0x57
github.com/nfx/go-htmltable.NewSlice[...]({0x71c310, 0xc00001a100?}, {0x7f3d447d5830?, 0xc00049c180?})
        GOPATH/pkg/mod/github.com/nfx/go-htmltable@v0.2.1/slice.go:17 +0x90
github.com/nfx/go-htmltable.NewSliceFromResponse[...](0x6b0fb8?)
        GOPATH/pkg/mod/github.com/nfx/go-htmltable@v0.2.1/slice.go:30 +0x8c
github.com/nfx/go-htmltable.NewSliceFromURL[...]({0x6b0fb8, 0x407ab9?})
        GOPATH/pkg/mod/github.com/nfx/go-htmltable@v0.2.1/slice.go:43 +0xc8
main.main()
        HOME/intros/html-tables/main.go:23 +0x30
exit status 2
```

### Go version

`go1.19.1 linux/amd64`

**Comment by nfx** [comment]
Interesting, will try to look into it this weekend - thanks for the repro. Meanwhile, I would encourage you to make a PR: best way to write a special unit test and debug it step by step from VSCode. Something feels off because of colspans/rowspans

**Comment by nfx** [comment]
@j-mnr panic is removed, and now getting the parsing done right for these kinds of tables:

<img width="1184" alt="image" src="https://user-images.githubusercontent.com/259697/198896270-1a39a3aa-9862-483d-be44-34022b47d8ef.png">

**Comment by nfx** [comment]
Once you update to the latest version, please change `Fab (nm)` -> `Fab(nm)`
```go
func TestX(t *testing.T) {
	type Chipset struct {
...
		Fab         string `header:"Fab(nm)"`
		HT          string `header:"HT(MHz)"`
...
	}
	_, err := NewSliceFromURL[Chipset]("https://en.wikipedia.org/wiki/List_of_AMD_chipsets")
	if err != nil {
		t.Fatal(err)
	}
}
```

**Comment by nfx** [comment]
@j-mnr

**Comment by j-mnr** [comment]
Hey super great you got it sorted! :grin: Just to reiterate, at the time I submitted this issue, I needed a tool like this two weeks before that. I have no current use for it.

I was trying to do my part in helping out in a small way, because as it's said: 80% of the people will find an error and stop using the service while the other 20% will give feedback. So supposedly 8 others ran into this before I did and didn't bother.

If I need an HTML table scraper in the future I'll definitely keep your repo in mind. Cheers :beers:
